src/libertem/utils/threading.py

```diff
@@ -63,6 +62,51 @@ def set_numba_threads(n):
 
 @contextmanager
 def set_num_threads(n):
+    import threadpoolctl
     with threadpoolctl.threadpool_limits(n), set_fftw_threads(n),\
             set_torch_threads(n), set_numba_threads(n):
         yield
+
+
+@contextmanager
+def set_num_threads_env(n=1, set_numba=None):
+    """Set the maximum number of threads via environment variables.
+    Currently sets variables for MKL, OMP, OPENBLAS and NUMBA.
+
+    This is needed if you want to limit the number of threads that
+    are created at startup of those libraries, because the other
+    methods usually only limit the number of threads that are _used_,
+    not the number of threads that are created.
+
+    Parameters
+    ----------
+    n : int
+        The maximum number of threads
+    set_numba : bool, optional
+        Set the :code:`'NUMBA_NUM_THREADS'` environment variable. If None, determine if
+        Numba has been initialized and only set it if not.
+        Numba may throw a :class:`RuntimeError` if the environment is
+        altered after the number of threads has been set already.
+    """
+    if set_numba is None:
+        from numba.np.ufunc import parallel
+        set_numba = not parallel._is_initialized
```

Comment:
As we are accessing an implementation detail here, maybe we could replace this with the following?

```suggestion
        set_numba = not getattr(parallel, "_is_initialized", True)
```

Then, in case numba changes the implementation, we assume that numba is already initialized and don't trigger an error, possibly at the expense of "resource waste" by starting too many threads.

What do you think?

Reply:
Thank you, good point! :-)